io/io/src/TFile.cxx

```diff
@@ -396,6 +397,8 @@ TFile::TFile(const char *fname1, Option_t *option, const char *ftitle, Int_t com
    Bool_t devnull = kFALSE;
 
    if (!fname1 || !fname1[0]) {
+      fname1.find("://") != std::string::npos ?
```

Comment:
Note, this is the wrong 'branch' to make this test.  Here (line 400 to 401), the string is 'empty' (fname1 is either a nullptr or its first character is a `\0`.

Reply:
Oh, I see. Which branch would I have to push this to?

Reply:
I did not mean branch in the git sense but in the if-then-else sense.   I.e. the code where you currently have it will never do anything, because if the cpu takes this code path then the string is empty (and so will never have "://" in it).

Please make sure you compile the code and test it.  A simple test could be:
```
root [0] f = new TFile("doesnotexist.root")
Error in <TFile::TFile>: file /Users/pcanal/root_working/roottest/doesnotexist.root does not exist
(TFile *) 0x7ff721fcb620
root [1] f = new TFile("")
Error in <TFile::TFile>: file name is not specified
(TFile *) 0x7ff721d879a0
root [2] f = new TFile("https://root.cern.ch/files/atlasFlushed.root")
Error in <TFile::TFile>: file /Users/pcanal/root_working/roottest/files/atlasFlushed.root does not exist
(TFile *) 0x7ff701c44720
```

where the new behavior should be:

```
root [0] f = new TFile("doesnotexist.root")
Error in <TFile::TFile>: file /Users/pcanal/root_working/roottest/doesnotexist.root does not exist
(TFile *) 0x7ff721fcb620
root [1] f = new TFile("")
Error in <TFile::TFile>: file name is not specified
(TFile *) 0x7ff721d879a0
root [2] f = new TFile("https://root.cern.ch/files/atlasFlushed.root")
Error in <TFile::TFile>: To open a file with an explicit remote protocol ("https://root.cern.ch/files/atlasFlushed.root") please use TFile::Open
(TFile *) 0x7ff701c44720
```

Reply:
Ahh ok I see. I was having some issues building root locally, but I'll try